MFiles.VAF.Extensions.Tests/ExtensionMethods/ObjVerEx/TryGetPropertyText.cs

```diff
@@ -0,0 +1,206 @@
+
+using Moq;
+
+namespace MFiles.VAF.Extensions.Tests.ExtensionMethods.ObjVerEx
+{
+	/// <summary>
+	/// Tests <see cref="ObjVerExExtensionMethods.TryGetPropertyText(Common.ObjVerEx, out string, VAF.Configuration.MFIdentifier, string)"/>
+	/// </summary>
+	[TestClass]
+	public class TryGetPropertyText
+		: TestBaseWithVaultMock
+	{
+		#region Class for props params
+		public class PropertyDefTestEnvironment
+		{
+			public PropertyDefTestEnvironment(MFIdentifier prop, Common.ObjVerEx objverEx)
+			{
+				Ident = prop;
+				IsResolved = prop?.IsResolved ?? false;
+				IsAssigned = IsResolved && null != objverEx && objverEx.HasProperty(prop);
+				Value = IsResolved ? objverEx.GetPropertyText(prop) : null;
+			}
+
+			public MFIdentifier Ident;
+			public bool IsResolved;
+			public bool IsAssigned;
+			public string Value;
+
+			public override string ToString()
+			{
+				return $"\n- #PropDef: {Ident?.ID.ToString() ?? "null"}" +
+					$"\n- Resolved: {IsResolved}\n- Assigned: {IsAssigned}" +
+					$"\n- Value:    {(null == Value ? "null" : "\"" + Value + "\"")}";
+			}
+		}
+		#endregion
+
+		#region Initialized MFIdentifier and ObjVerEx objects for properties for testing
+		private Common.ObjVerEx objVerEx;
+		private const Common.ObjVerEx nullObjVerEx = null;
+		private const int idCustomProp = 1111;
+		private const string aliasCustomProp = "PD.Test";
+		private PropertyDefTestEnvironment envNull;
+		private PropertyDefTestEnvironment envNameOrTitle;
+		private PropertyDefTestEnvironment envKeywords;
+		private PropertyDefTestEnvironment envMessageID;
+		private PropertyDefTestEnvironment envCustomProp;
+		private PropertyDefTestEnvironment envNotResolved;
+		private List<PropertyDefTestEnvironment> ListEnvironments => new List<PropertyDefTestEnvironment>()
+		{
+			envNull, envNameOrTitle, envKeywords, envMessageID, envCustomProp, envNotResolved
+		};
+		#endregion
+
+		/// <summary>
+		/// Ensures that a null <see cref="Common.ObjVerEx"/> reference throws an <see cref="ArgumentNullException"/>.
+		/// </summary>
+		[TestMethod]
+		[ExpectedException(typeof(ArgumentNullException))]
+		public void TryGetPropertyText_ThrowsIfNullObjVerEx()
+			=> _ = nullObjVerEx.TryGetPropertyText(out _, envNameOrTitle.Ident);
+
+		/// <summary>
+		/// Ensures that not resolved <see cref="MFIdentifier"/> objects return <see cref="true"/>
+		/// </summary>
+		[TestMethod]
+		public void TryGetPropertyText_ResultForNullIdentifier()
+			=> Assert.IsFalse(objVerEx.TryGetPropertyText(out _, envNull.Ident));
+
+		/// <summary>
+		/// Ensures that not resolved <see cref="MFIdentifier"/> objects return <see cref="true"/>
+		/// </summary>
+		[TestMethod]
+		public void TryGetPropertyText_ResultForNotResolved()
+			=> Assert.IsFalse(objVerEx.TryGetPropertyText(out _, envNotResolved.Ident));
+
+		/// <summary>
+		/// Check all return values for all entries
+		/// </summary>
+		[TestMethod]
+		public void TryGetPropertyText_CheckOutputForInitializedEntries()
```

Comment:
For consistency, I would recommend the use of the `[DynamicData]` approach to provide data for additional testing.  At first glance this may mean some duplication in the setup of the tests (I notice that multiple tests use the same data such as envNotResolved).

This is not mandatory before merge, but may be something that I undertake after merge.

Reply:
I've just spend some time on testing how to test - and finally left a version which works and covers hopefully all cases.
If you have a better solution I would be happy to take a look at it afterwards.